File: t.py
import os
import re

directory = "c:\\Users\\perpl\\aequitaslabs-new"

# Regex for replacing SVG logo with new image
logo_pattern = re.compile(r'<div class="logo-mark">\s*<svg.*?</svg>\s*</div>', re.DOTALL)
new_logo = '<div class="logo-mark"><img src="assets/aequitas-pfp-400x400.png" alt="AequitasLabs Logo" style="width:100%;height:100%;object-fit:contain;"></div>'

new_logo_abs = '<div class="logo-mark"><img src="https://aequitaslabs.xyz/assets/aequitas-pfp-400x400.png" alt="AequitasLabs Logo" style="width:100%;height:100%;object-fit:contain;"></div>'

# Regex for favicon
favicon_pattern = re.compile(r'<link rel="icon" type="image/svg\+xml" href="data:image/svg\+xml,[^"]*"/>')
new_favicon = '<link rel="icon" type="image/png" href="assets/aequitas-pfp-400x400.png"/>'
new_favicon_abs = '<link rel="icon" type="image/png" href="https://aequitaslabs.xyz/assets/aequitas-pfp-400x400.png"/>'


for root, dirs, files in os.walk(directory):
    if '.git' in root or 'node_modules' in root:
        continue
    for file in files:
        if file.endswith('.html') or file.endswith('.tsx'):
            file_path = os.path.join(root, file)
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()

            is_dapp = "dapp" in root or "dapp\\pages" in root

            original_content = content

            if not is_dapp:
                content = logo_pattern.sub(new_logo, content)
                content = favicon_pattern.sub(new_favicon, content)
            else:
                # for dapp
                content = favicon_pattern.sub(new_favicon_abs, content)
                # also replace the custom inline dummy logo
                dummy_logo_pattern = re.compile(r'<div style={{ width: \'24px\', height: \'24px\', border: \'1px solid #C9A84C.*?</div\s*>', re.DOTALL)
                content = dummy_logo_pattern.sub('<img src="https://aequitaslabs.xyz/assets/aequitas-pfp-400x400.png" alt="AequitasLabs Logo" style={{ width: \'24px\', height: \'24px\', objectFit: \'contain\' }} />', content)


            if content != original_content:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Updated {file_path}")

File: t.js
const fs = require('fs');
const path = require('path');

const directory = 'c:\\Users\\perpl\\base-new';

const logoPattern = /<span style="color:#0052FF;font-weight:800">Base</span>';

const faviconPattern = /<link rel="icon" type="image\/svg\+xml" href="data:image\/svg\+xml,[^"]*"\/>/g;
const newFavicon = '<link rel="icon" type="image/png" href="assets/base-pfp-400x400.png"/>';
const newFaviconAbs = '<link rel="icon" type="image/png" href="https://base.xyz/assets/base-pfp-400x400.png"/>';

function walkDir(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach((file) => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            if (!file.includes('.git') && !file.includes('node_modules')) {
                results = results.concat(walkDir(file));
            }
        } else {
            if (file.endsWith('.html') || file.endsWith('.tsx')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walkDir(directory);
files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;
    let isDapp = file.includes('dapp') || file.includes('dapp\\pages');

    if (!isDapp) {
        content = content.replace(logoPattern, newLogo);
        content = content.replace(faviconPattern, newFavicon);
    } else {
        content = content.replace(faviconPattern, newFaviconAbs);
        const dummyLogoPattern = /<div style={{ width: '24px', height: '24px', border: '1px solid #C9A84C[\s\S]*?<\/div\s*>\n\s*<\/div>/g;
        // Wait, the dummy logo actually has an inner div too:
        //           <div style={{ width: '24px', height: '24px', border: '1px solid #C9A84C', ... }}>
        //             <div style={{ width: '8px', height: '8px', background: '#C9A84C', ... }}></div>
        //           </div>
        const dummyLogoPatt2 = /<div style={{ width: '24px', height: '24px', border: '1px solid #C9A84C[\s\S]{1,300}?<\/div>/g;
        content = content.replace(dummyLogoPatt2, '');
    }

    if (content !== original) {
        fs.writeFileSync(file, content, 'utf8');
        console.log('Updated ' + file);
    }
});
